<launch>

  <arg name="fcu_url" default="/dev/pixhawk:921600" />
  <arg name="gcs_url" default="" />
  <arg name="tgt_system" default="" />
  <arg name="tgt_component" default="" />
  <arg name="pluginlists_yaml" value="$(find mrs_uav_general)/config/px4_pluginlists.yaml" />
  <arg name="config_yaml" value="$(find mrs_uav_general)/config/px4_config.yaml" />
  <arg name="log_output" default="screen" />
  <arg name="fcu_protocol" default="v2.0" />
  <arg name="respawn_mavros" default="false" />

  <arg name="UAV_NAME" default="$(env UAV_NAME)" />

  <group ns="$(arg UAV_NAME)">

    <node pkg="mavros" type="mavros_node" name="mavros" required="$(eval not respawn_mavros)" clear_params="true" output="$(arg log_output)" respawn="$(arg respawn_mavros)">

      <param name="fcu_url" value="$(arg fcu_url)" />
      <param name="gcs_url" value="$(arg gcs_url)" />
      <param name="target_system_id" value="$(arg tgt_system)" />
      <param name="target_component_id" value="$(arg tgt_component)" />
      <param name="fcu_protocol" value="$(arg fcu_protocol)" />

      <param name="distance_sensor/garmin/frame_id" value="$(arg UAV_NAME)/garmin" />
      <param name="local_position/frame_id" value="$(arg UAV_NAME)/pixhawk_origin" />
      <param name="local_position/tf/child_frame_id" value="$(arg UAV_NAME)/fcu" />
      <param name="imu/frame_id" value="$(arg UAV_NAME)/fcu" />
      <param name="global_position/frame_id" value="$(arg UAV_NAME)/utm_origin" />
      <param name="global_position/child_frame_id" value="$(arg UAV_NAME)/fcu" />

      <!-- load blacklist, config -->
      <rosparam command="load" file="$(arg pluginlists_yaml)" />
      <rosparam command="load" file="$(arg config_yaml)" />

      <!-- prevent all uavs from publishing to the same diagnostics topic -->
      <remap from="/diagnostics" to="/$(arg UAV_NAME)/diagnostics" />

    </node>

  </group>

</launch>
